### PR TITLE
feat(ci): run flux integration tests as its own job

### LINKED ISSUE #20888: Run Flux Acceptance Tests via CI in InfluxDB
We need to ensure that Flux Acceptance Tests are run as a step in the CI process for InfluxDB.

DOD:
* If a Flux Acceptance Tests fails it is reported in CI and prevents the merge
* The test that fails can be determined from the CI run log output

### PR DESCRIPTION
**Pull request by jsternberg**
Fixes #20888.

### CODE REVIEW

File: etc/test-flux.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+log() {
+  local now
+  now=$(date '+%Y/%m/%d %H:%M:%S')
+  echo "[${now}]" "$@"
+}
+
+determine_flux_revision() {
+  local version revision
+  version=$(go list -m -f '{{.Version}}' github.com/influxdata/flux)
```
Comment (mark-rushakoff) [inline etc/test-flux.sh:13]: This should have been `"$GO"` like on L31 (I overlooked that in the other PR). Surrouding quotes on the off-chance that the environment variable contains whitespace.

**Review by mark-rushakoff — APPROVED** [review]

**Review by wolffcm — APPROVED** [review]

**Review by wolffcm — APPROVED** [review]
Looks good to me.

Should someone from the Edge Team have a look as well? @lesam maybe?

**Comment by lesam** [comment]
How is this different from `TestFluxEndToEnd` ? Do we still need that one?

**Comment by jsternberg** [comment]
It's intended to replace that one in the future but it requires a transition.

**Review by lesam — APPROVED** [review]
looks good - is there an issue tracking the `TestFluxEndToEnd` removal? Would save some CPU (and maybe time) for CI checks

**Comment by jsternberg** [comment]
I'm not sure if we have an issue for that yet.

**Pull request merged**